models/data.go

```diff
@@ -0,0 +1,195 @@
+package models
+
+import (
+	"crypto/sha256"
+	"encoding"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+)
+
+var ErrContentApplied = errors.New("data has been submitted")
+
+type DataType uint8
+
+const (
+	Simple DataType = iota
+	JSON
+	Audio
+	Video
+)
+
+type Hash32 [32]byte
+
+func (h Hash32) String() string {
+	return hex.EncodeToString(h[:])
+}
+
+func (h Hash32) IsEmpty() bool {
+	return h == Hash32{}
+}
+
+type Data interface {
+	encoding.BinaryMarshaler
+	encoding.BinaryUnmarshaler
+
+	CachedHash() Hash32
```

Comment:
Let's rename this method to ID and remove IsCorrect. Don't worry about changed hash of file, in future you will generate it randomly=)